app/src/main/java/com/example/myapplication/MainActivity.kt

```diff
@@ -1,17 +1,125 @@
-    Successful registration (new user is present in CredManager)
-    Case Insensitive in email
-
-    Apply it to SignUpDetailsActivity
-    After successful registration, go to CreateAccount activity
-    */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val sharedPreferences = getSharedPreferences("user_prefs", MODE_PRIVATE)
+        val isLoggedIn = sharedPreferences.getBoolean("is_logged_in", false)
+
+        if (isLoggedIn) {
+            loadMainContent()
+        } else {
+            if (savedInstanceState == null) {
+                supportFragmentManager.beginTransaction()
+                    .replace(R.id.fragmentContainerView, CreateAccountFragment())
+                    .commit()
+            }
+        }
+        val logoutButton: Button = findViewById(R.id.logoutButton)
+        logoutButton.setOnClickListener {
+            val editor = sharedPreferences.edit()
+            editor.putBoolean("is_logged_in", false)
+            editor.apply()
+
+            navigateToCreateAccountFragment()
+
+            findViewById<RecyclerView>(R.id.recyclerView).visibility = View.GONE
+            findViewById<Button>(R.id.logoutButton).visibility = View.GONE
+        }
+    }
+
+    fun loadMainContent() {
+        val recyclerView: RecyclerView = findViewById(R.id.recyclerView)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+
+        val recipes = listOf(
+            Recipe(1, "Takoyaki", R.drawable.takoyaki),
+            Recipe(2, "Tempura", R.drawable.tempura),
+            Recipe(3, "Ramen", R.drawable.ramen)
+        )
+
+        val adapter = RecipeAdapter(
+            recipes,
+            itemClickListener = { recipe ->
+                val intent = Intent(this, RecipeDetailActivity::class.java).apply {
+                    putExtra("RECIPE_ID", recipe.id)
+                    putExtra("RECIPE_TITLE", recipe.title)
+                    putExtra("RECIPE_IMAGE", recipe.imageResId)
+                }
+                startActivity(intent)
+            },
+            likeClickListener = { recipe ->
+                Toast.makeText(this, "Liked: ${recipe.title}", Toast.LENGTH_SHORT).show()
+            },
+            shareClickListener = { recipe ->
+                Toast.makeText(this, "Shared: ${recipe.title}", Toast.LENGTH_SHORT).show()
+            }
+        )
+
+        recyclerView.adapter = adapter
+    }
+
+    private fun navigateToCreateAccountFragment() {
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.fragmentContainerView, CreateAccountFragment())
+            .commit()
+    }
+}
+
+data class Recipe(
+    val id: Int,
+    val title: String,
+    val imageResId: Int
+)
+
+class RecipeAdapter(
+    private val recipes: List<Recipe>,
+    private val itemClickListener: (Recipe) -> Unit,
```

Comment:
That's one way to do it.
In bigger adapters that will get overwhelming though (it almost already is), then using interfaces would be more readable imo